File: src/main/java/net/shotbow/interestingfish/InterestingFish.java
package net.shotbow.interestingfish;

import net.shotbow.interestingfish.config.InterestingConfig;
import net.shotbow.interestingfish.listeners.FishListener;
import net.shotbow.interestingfish.listeners.InteractListener;
import org.bukkit.Bukkit;
import org.bukkit.plugin.java.JavaPlugin;

/**
 * User: Austin
 * Date: 5/18/13
 * Time: 12:43 AM
 * (c) lazertester
 */
public class InterestingFish extends JavaPlugin
{


    private InterestingConfig config;

    public static void log(String message)
    {
        Bukkit.getLogger().info("[InterestingFish] " + message);
    }

    public void onEnable()
    {
        config = new InterestingConfig(getDataFolder().getPath());
        config.initialize();
        final FishInfoFactory fishInfoFactory = new FishInfoFactory(config);
        Bukkit.getPluginManager().registerEvents(new FishListener(fishInfoFactory, config), this);
        Bukkit.getPluginManager().registerEvents(new InteractListener(), this);
        log("InterestingFish is enabled!");
    }
}

File: src/main/kotlin/net/shotbow/interestingfish/InterestingFish.kt
package net.shotbow.interestingfish

import com.uchuhimo.konf.Config
import com.uchuhimo.konf.source.yaml
import com.uchuhimo.konf.toValue
import net.shotbow.interestingfish.config.InterestingConfig
import net.shotbow.interestingfish.listeners.FishListener
import net.shotbow.interestingfish.listeners.InteractListener
import org.bukkit.Bukkit
import org.bukkit.plugin.java.JavaPlugin
import java.io.File

class InterestingFish : JavaPlugin() {
    override fun onEnable() {
        saveDefaultConfig() // make sure we have at least a default config file

        val config = Config().from.yaml.file(File(dataFolder, "config.yml")).toValue<InterestingConfig>()
        log(config.toString())
        val fishInfoFactory = FishInfoFactory(config)

        Bukkit.getPluginManager().registerEvents(FishListener(fishInfoFactory, config), this)
        Bukkit.getPluginManager().registerEvents(InteractListener(config), this)
        log("InterestingFish is enabled!")
    }

    companion object {
        fun log(message: String) {
            Bukkit.getLogger().info("[InterestingFish] $message")
        }
    }
}
